### PR TITLE
Touch up CI, broken by github installing GHC 9.2.1

### PR DESCRIPTION
**Pull request by Mikolaj**

### CODE REVIEW

File: .github/workflows/quick-jobs.yml [hunk 1]
```diff
@@ -62,6 +66,10 @@ jobs:
         with:
           path: ~/.cabal/store
           key: linux-store-doctest
+      - name: ghcup
+        run: |
+          ghcup install ghc recommended
+          ghcup set recommended
```
Comment (jneira) [inline .github/workflows/quick-jobs.yml:72]: you can do it in one step with `ghcup install ghc recommended --set`

**Review by jneira — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/quick-jobs.yml [hunk 2]
```diff
@@ -23,11 +23,15 @@ jobs:
       - name: Set PATH
         # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#adding-a-system-path
         run: |
-          echo "$HOME/.cabal/bin" >> $GITHUB_PATH
+          echo "$HOME/.cabal/bin:$HOME/.ghcup/bin" >> $GITHUB_PATH
```
Comment (jneira) [inline .github/workflows/quick-jobs.yml:26]: is not $HOME/.ghcup/bin already in PATH?

Comment (Mikolaj) [reply .github/workflows/quick-jobs.yml:26]: I probably panicked. Will remove and see.

File: .github/workflows/quick-jobs.yml (hunk 1 above)
Comment (Mikolaj) [reply .github/workflows/quick-jobs.yml:72]: Noted. Let's keep it more verbose for now, while we start the overhaul.

**Comment by Mikolaj** [comment]
In the end, I had to remove most of my changes, because most of the jobs use old tools that can't get GHC 8.10.7 (containers, possibly curl for Mac).

**Comment by Mikolaj** [comment]
BTW, my Matrix died in case somebody is looking for me. @emilypi: I don't have anything for today; do we meet?

**Comment by Mikolaj** [comment]
This PR has shrunk a lot and the CI is broken as we speak, so let me merge with only one review. CI here passes 100%.

**Comment by Mikolaj** [comment]
mergifyIO backport 3.6

**Comment by Mikolaj** [comment]
@Mergifyio backport 3.6

**Pull request merged**